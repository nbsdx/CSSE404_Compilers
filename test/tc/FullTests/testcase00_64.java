/*
 * Testcase designed to test statements.
 * This should not pass, and the compiler should give errors to the effect of
 * "type mismatch" and "undefined variable at lexical scope",
 * and may adjust them for various statement types.
 */

class Test7 {
	public static void main(String[] args) {
		int x = 5;
		while(x < true) {
			x = x + 1;
			y = y + 3;
		}

		{
			x = 5;
			{
				int y = 5;
				y = x;
			}
			x = y;
		}

		if (x < 10) {
			System.out.println(false);
		} else {
		
		}
	}
}

// vim:filetype=java:textwidth=80
