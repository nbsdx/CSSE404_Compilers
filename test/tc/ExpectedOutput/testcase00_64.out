Right argument of type boolean does not match expected type int for operator LT
Variable y is not declared.
Left argument of type null does not match expected type int for operator PLUS
No variable named y exists in the current scope.
Variable y is not declared.
Cannot assign type null to variable x of type int
In MiniJava, System.out.println only takes an int.  The expression has type boolean
